Source/JavaScriptCore/interpreter/CalleeBits.h

```diff
@@ -117,7 +117,12 @@ class CalleeBits {
     static void* boxNativeCallee(NativeCallee* callee)
     {
 #if USE(JSVALUE64)
-        CalleeBits result { static_cast<int64_t>((bitwise_cast<uintptr_t>(callee) - lowestAccessibleAddress()) | JSValue::NativeCalleeTag) };
+        auto bits = bitwise_cast<uintptr_t>(callee);
+#if CPU(ARM64)
+        // NativeCallees are sometimes stored in ThreadSafeWeakOrStrongPtr so we need to strip the top byte on ARM64.
+        bits &= (std::numeric_limits<uintptr_t>::max() >> CHAR_BIT);
+#endif
+        CalleeBits result { static_cast<int64_t>((bits - lowestAccessibleAddress()) | JSValue::NativeCalleeTag) };
```

Comment:
This boxing is done in AssemblyHelpers (JIT) and LLInt (WebAssembly.asm and IPInt code). Do we need to change them too?

Reply:
Yeah, we probably should since in the future we could hold those callees weakly. I think in this patch it's fine since only BBQCallee/OSREntryCallee are stored weakly but if we ever started holding JSToWasmCallees weakly we'd see a crash.